File: android/src/main/java/iyegoroff/multisegmentedcontrol/MultiSegmentedControlManager.java
package iyegoroff.multisegmentedcontrol;

import android.graphics.Color;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.uimanager.ViewGroupManager;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.facebook.react.uimanager.ThemedReactContext;

import javax.annotation.Nonnull;

@ReactModule(name = MultiSegmentedControlManager.REACT_CLASS)
public class MultiSegmentedControlManager extends ViewGroupManager<MultiSegmentedControl> {

  static final String REACT_CLASS = "MSCMultiSegmentedControl";

  @Override
  public @Nonnull String getName() {
    return REACT_CLASS;
  }

  @Override
  public @Nonnull MultiSegmentedControl createViewInstance(@Nonnull ThemedReactContext reactContext) {
    return new MultiSegmentedControl(reactContext);
  }

  @ReactProp(name = "values")
  public void setValues(MultiSegmentedControl view, ReadableArray values) {
    view.setValues(values);
  }

  @ReactProp(name = "selectedIndices")
  public void setSelectedIndices(MultiSegmentedControl view, ReadableArray selectedIndices) {
    view.setSelectedIndices(selectedIndices);
  }

  @ReactProp(name = "tintColor", customType = "Color", defaultInt = Color.BLUE)
  public void setTintColor(MultiSegmentedControl view, int tintColor) {
    view.setTintColor(tintColor);
  }

  @ReactProp(name = "enabled", defaultBoolean = true)
  public void setEnabled(MultiSegmentedControl view, boolean enabled) {
    view.setEnabled(enabled);
  }

  @ReactProp(name = "isSingle")
  public void setIsSingle(MultiSegmentedControl view, boolean isSingle) {
    view.setIsSingle(isSingle);
  }

  @ReactProp(name = "maxSelected")
  public void setMaxSelected(MultiSegmentedControl view, int maxSelected) {
    view.setMaxSelected(maxSelected);
  }

  @ReactProp(name = "minSelected")
  public void setMinSelected(MultiSegmentedControl view, int minSelected) {
    view.setMinSelected(minSelected);
  }

  @ReactProp(name = "dividerColor", customType = "Color", defaultInt = Color.GRAY)
  public void setDividerColor(MultiSegmentedControl view, int dividerColor) {
    view.setDividerColor(dividerColor);
  }

  @ReactProp(name = "borderRadius")
  public void setBorderRadius(MultiSegmentedControl view, float borderRadius) {
    view.setBorderRadius(borderRadius);
  }

  @ReactProp(name = "textStyle")
  public void setTextStyle(MultiSegmentedControl view, ReadableMap textStyle) {
    view.setTextStyle(textStyle);
  }

  @ReactProp(name = "selectedTextStyle")
  public void setSelectedTextStyle(MultiSegmentedControl view, ReadableMap selectedTextStyle) {
    view.setSelectedTextStyle(selectedTextStyle);
  }

  @ReactProp(name = "elevation")
  public void setElevation(@Nonnull MultiSegmentedControl view, float elevation) {
    view.setCardElevation(elevation);
  }
}

File: android/src/main/java/iyegoroff/multisegmentedcontrol/MultiSegmentedControlManager.kt
package iyegoroff.multisegmentedcontrol

import android.graphics.Color
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.ViewGroupManager
import com.facebook.react.uimanager.annotations.ReactProp

@ReactModule(name = MultiSegmentedControlManager.reactClass)
class MultiSegmentedControlManager : ViewGroupManager<MultiSegmentedControl>() {

  companion object {
    const val reactClass = "MSCMultiSegmentedControl"
  }

  override fun getName(): String {
    return reactClass
  }

  override fun createViewInstance(reactContext: ThemedReactContext): MultiSegmentedControl {
    return MultiSegmentedControl(reactContext)
  }

  @ReactProp(name = "values")
  fun setValues(view: MultiSegmentedControl, values: ReadableArray) {
    view.setValues(values)
  }

  @ReactProp(name = "selectedIndices")
  fun setSelectedIndices(view: MultiSegmentedControl, selectedIndices: ReadableArray) {
    view.setSelectedIndices(selectedIndices)
  }

  @ReactProp(name = "tintColor", customType = "Color", defaultInt = Color.BLUE)
  fun setTintColor(view: MultiSegmentedControl, tintColor: Int) {
    view.setTintColor(tintColor)
  }

  @ReactProp(name = "enabled", defaultBoolean = true)
  fun setEnabled(view: MultiSegmentedControl, enabled: Boolean) {
    view.isEnabled = enabled
  }

  @ReactProp(name = "isSingle")
  fun setIsSingle(view: MultiSegmentedControl, isSingle: Boolean) {
    view.setIsSingle(isSingle)
  }

  @ReactProp(name = "maxSelected")
  fun setMaxSelected(view: MultiSegmentedControl, maxSelected: Int) {
    view.setMaxSelected(maxSelected)
  }

  @ReactProp(name = "minSelected")
  fun setMinSelected(view: MultiSegmentedControl, minSelected: Int) {
    view.setMinSelected(minSelected)
  }

  @ReactProp(name = "dividerColor", customType = "Color", defaultInt = Color.GRAY)
  fun setDividerColor(view: MultiSegmentedControl, dividerColor: Int) {
    view.setDividerColor(dividerColor)
  }

  @ReactProp(name = "borderRadius")
  fun setBorderRadius(view: MultiSegmentedControl, borderRadius: Float) {
    view.setBorderRadius(borderRadius)
  }

  @ReactProp(name = "textStyle")
  fun setTextStyle(view: MultiSegmentedControl, textStyle: ReadableMap) {
    view.setTextStyle(textStyle)
  }

  @ReactProp(name = "selectedTextStyle")
  fun setSelectedTextStyle(view: MultiSegmentedControl, selectedTextStyle: ReadableMap) {
    view.setSelectedTextStyle(selectedTextStyle)
  }

  @ReactProp(name = "elevation")
  override fun setElevation(view: MultiSegmentedControl, elevation: Float) {
    view.setCardElevation(elevation)
  }
}